### PR TITLE
Changed type of ignoreInVersions in visWidgets

### PR DESCRIPTION
**Pull request by GermanBluefox**

### CODE REVIEW

**Comment by foxriver76** [comment]
is it correct like this or recent change? If it was a recent design change, then we should avoid having properties with free chooseable names and fixed names on the same level.

**Comment by GermanBluefox** [comment]
> is it correct like this or recent change? If it was a recent design change, then we should avoid having properties with free chooseable names and fixed names on the same level.
> 
> Also in the schema it is not on level of visWidgets property but on the level of the individual widgetSet.. so what is now if I want to provide one widget set exclusive for vis1 another exclusive for vis2? This would no longer be configurable..

I changed a bit

File: packages/types-dev/objects.d.ts [hunk 1]
```diff
@@ -677,7 +675,8 @@ declare global {
             unsafePerm?: true;
             /** The available version in the ioBroker repo. */
             version: string;
-            visWidgets?: Record<string, VisWidget>;
+            /** The vis widget does not support the listed major versions of vis */
+            visWidgets?: Record<string, VisWidget> | { ignoreInVersions?: number[] };
```
Comment (foxriver76) [inline packages/types-dev/objects.d.ts:679]: Now you can either define via widgets or ignore I do not think it is the intention. See also my previous comment.

**Review by foxriver76 — CHANGES_REQUESTED** [review]

File: packages/types-dev/objects.d.ts (hunk 1 above)
Comment (foxriver76) [reply packages/types-dev/objects.d.ts:686]: Okay, please confirm: 

Valid: 
```json
....
"visWidgets": {
    "ignoreInVersions": [2]
}
.... 
```

Valid: 
....
"visWidgets": {
    "mySuperWidget": { ... }
}
.... 
```

Invalid:
....
"visWidgets": {
    "mySuperWidget": { ... },
    "ignoreInVersions": [2]
},
.... 
```

Maybe add some type tests to `types-public` package in `index.test-d.ts` to ensure it works like intended.

**Comment by foxriver76** [comment]
as we have discussed in one of the core meetings the schema will stay and implementation should follow it

**Pull request closed without merging**